src/foam/core/CompoundException.js

```diff
@@ -98,21 +96,36 @@ foam.CLASS({
     {
       name: 'add',
       args: [{ name: 't', javaType: 'Throwable' }],
-      javaCode: 'getExceptions().add(t);'
+      javaCode: `
+        for ( var e : t.getSuppressed() ) {
+          add(e);
+        }
+
+        if ( t.getSuppressed().length == 0 ) {
```

Comment:
Maybe we only want to flatten it if it is a compound exception. Other code may throw suppressing exceptions, in which case we probably want the extra information from the top one. So maybe:
if ( CompoundException ) { add all suppressed } else { add(t) }
Thoughts?

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/4767/commits/da5134435708559c452af6e1a55ba52b492d49ff. Thanks.